### PR TITLE
Unit tests cleanup

### PR DESCRIPTION
**Pull request by a-urth**
There are two major things within this PR
- ordering for string items in `FormatDSN` - it doesn't affect any existing functionality although it makes tests based on URL comparison much easier since items' order within variadic query args will be consistent
- unit tests cleanup - use `testify` library to make assertions shorter and more verbose, use `t.Cleanup` instead of `defer`

### CODE REVIEW

**Comment by a-urth** [comment]
Integration test if failing, thats the same one as in my other PR. I have a feeling that behaviour might have change in trinodb used for integration tests and was not reflected in test itself.

**Comment by hashhar** [comment]
Indeed, master fails on latest image too - https://github.com/trinodb/trino-go-client/pull/28.

**Comment by hashhar** [comment]
Closing and re-opening to get CI to re-run.

**Review by hashhar — COMMENTED** [review]
Revert the dependency changes if not required (since it seems it just replaces stdlib tests with testify?)

Looks good otherwise.

**Comment by a-urth** [comment]
Reverted version bump for testify lib, since it was already used in the project although not directly.

@hashhar as for tests, this lib is very useful, since you don't need to check for error (or other condition) and then fail test. Tests functionally the same, but just look way cleaner

File: trino/trino_test.go [hunk 1]
```diff
@@ -418,28 +412,27 @@ func TestTypeConversion(t *testing.T) {
...
-			if err != nil {
-				t.Fatal(err)
-			}
-			if !reflect.DeepEqual(v, tc.ExpectedGoValue) {
-				t.Fatalf("unexpected data from sample:\nhave %+v\nwant %+v", v, tc.ExpectedGoValue)
-			}
+			require.NoError(t, err)
+
+			require.Truef(t,
+				reflect.DeepEqual(v, tc.ExpectedGoValue),
```
Comment (nineinchnick) [inline trino/trino_test.go:434]: Could you use `require.Equal()` here too? It seems to fall back to `DeepEqual` anyway: https://github.com/stretchr/testify/blob/master/assert/assertions.go#L65

**Review by nineinchnick — COMMENTED** [review]
I don't mind the explicit conditions in tests, but I think testify would produce better error messages, adding unexpected errors and expected vs actual values.

But if we were to switch to testify, we should be consistent, and use it in all tests, including integration tests.

**Comment by nineinchnick** [comment]
This PR addresses #11

**Comment by hashhar** [comment]
Agreed, let's extract a commit / PR though which adds the `sort` to ensure tests always get predictable ordering.

And keep the testify change in a separate commit / PR.

File: trino/trino_test.go (hunk 1 above)
Comment (a-urth) [reply trino/trino_test.go:434]: Sure thing, will do.

**Comment by hashhar** [comment]
Splitting into commits is also fine, doesn't have to be a separate PR. It's very difficult to notice in future that something functional changed (like sorting values) in a large diff where it looks like the only change is moving from stdlib tests to testify.

**Comment by a-urth** [comment]
@hashhar I've changed PR to have 2 main commits (and 1 additional from review):
- one with change to order variadic query args
- another one with tests overhaul

**Review by hashhar — APPROVED** [review]
Thanks. Changes look good to me.

Can you squash the last 2 commits into one (so that in the end there are only 2 commits)?

I'll merge once CI finishes.

**Comment by a-urth** [comment]
@hashhar i've joined last two commits with unit tests' restructure and review comments into one

**Comment by hashhar** [comment]
@a-urth Thanks. Looks good. Can we add some detail to second commit message like `Migrate test assertions to github.com/stretchr/testify/assert`?

**Comment by hashhar** [comment]
See https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#git-merge-strategy

**Review by hashhar — APPROVED** [review]

**Comment by hashhar** [comment]
Merged. Thanks for the improvements. 🎉

**Pull request merged**